Runtime/Managers/EditMediator.cs

```diff
@@ -271,8 +275,21 @@ private async Task LoadObjectsState(JObject requestPayload)
                         TypeHelpers.NotNull(objectState["material"]["texture"]);
                 })
                 .GroupBy(objectState => objectState["material"]["id"].ToObject<int>());
-            await ApplyMaterialsToSceneObjects(objectStatesByMaterial);
+            TaskResult materialWasChanged = await ApplyMaterialsToSceneObjects(objectStatesByMaterial);
 
+            TaskResult finalResult = TaskResult.CombineResults(materialWasChanged, colorWasChanged);
+            if (!finalResult.isSuccess)
+            {
+                originalSceneController?.RestoreOriginalScene();
```

Comment:
same as before, I'd prefer to crash everything and burn the whole place down, than going inadvertently about this possible step-skipping. 
```suggestion
                originalSceneController.RestoreOriginalScene();
```